src/stableswapCalculator.ts

```diff
@@ -317,10 +342,20 @@ export class StableswapCalculator implements SwapCalculator {
     liqA: bigint,
     liqB: bigint,
     amountDeposited: bigint,
+    saveIterations = true,
   ): bigint {
     const precision = BigInt(this.stableswapParams.precision);
     const amplifier = this.getAmplifier();
-    const invariant = getInvariant(liqA, liqB, amplifier, precision);
+
+    const [invariant, invariantIterations] = getInvariant(
+      liqA,
+      liqB,
+      amplifier,
+      precision,
+    );
+    if (saveIterations) {
+      this.invariantIterations = invariantIterations;
+    }
```

Comment:
what's the reason we don't save iterations always?

Reply:
`getPrice` uses `getSwapGrossAmountReceived` and can overwrite the iterations.